packages/coil-extension/src/background/services/Stream.ts

```diff
@@ -198,29 +207,40 @@ export class Stream extends EventEmitter {
           await timeout(1000)
         }
       } catch (e) {
+        if (plugin) await plugin.disconnect()
         const { ilpReject } = e
         if (
           btpToken &&
           ilpReject &&
           ilpReject.message === 'exhausted capacity.' &&
           ilpReject.data.equals(await sha256(Buffer.from(btpToken)))
         ) {
-          this._debug('anonymous token exhausted; retrying, err=%s', e.message)
+          this._debug(
+            'anonymous token exhausted; retrying soon, err=%s',
+            e.message
+          )
           this._anonTokens.removeToken(btpToken)
+          payScheduler.onSent()
+          await payScheduler.wait()
```

Comment:
how come this wait only happens in the catch branch? is it because we run the try portion all the way until we exhaust the token?

Reply:
Yes. The `catch` block is always called -- either because of an actual error or (in the success case) because the token was exhausted.